ipn/local.go

```diff
@@ -435,7 +435,7 @@ func (b *LocalBackend) updateDNSMap(netMap *controlclient.NetworkMap) {
 	if netMap == nil {
 		return
 	}
-	dnsMap := &tsdns.Map{DomainToIP: make(map[string]netaddr.IP)}
+	domainToIP := make(map[string]netaddr.IP)
```

Comment:
for large domains with a lot of nodes, this might be a lot of garbage per netmap update, repeatedly making these maps that are identical between updates. this is an alloc here, and is the `.ipn.dev` concat.

Does something else filter out the calls to `updateDNSMap` so it's only called on changes?

Reply:
Not currently. Since netmap diffing happens anyway, I could only `updateDNSMap` if the netmap diff is not empty, which will probably eliminate most of the unnecessary updates. Is that reasonable, or should there be more precise logic?

Reply:
Yeah, that's reasonable for now. Later we'll also want to skip updating the DNS map if just e.g. endpoints change (something that's not a name or tailscale IP)

Reply:
I've done this. Would appreciate a quick review very much, since incorrectly missing a filter update would be pretty terrible. I could just keep non-DNS related changes out, but it made most sense to me to capture those related things as well.